system/doc/reference_manual/expressions.xml

```diff
@@ -836,6 +836,28 @@ number &lt; atom &lt; reference &lt; fun &lt; port &lt; pid &lt; tuple &lt; map
       depending on the size of the float because otherwise comparison of large
       floats and integers would lose their transitivity.</p>
 
+    <p>The term equivalence operators, <c>=:=</c> and <c>=/=</c>, return
+      whether two terms are indistinguishable. While the other operators
+      consider the same <em>numbers</em> equal even when their types differ
+      (<c>1 == 1.0</c> is true), the term equivalence operators return whether
+      there exists any function that can tell their arguments apart.</p>
+
+    <p>For example, while the terms <c>0</c> and <c>0.0</c> represent the same
+      <em>number</em>, we can tell them apart by using the <c>is_integer/1</c>
+      function. Hence, <c>=:=</c> and <c>=/=</c> consider them different.</p>
+
+    <p>Furthermore, the terms <c>0.0</c> and <c>-0.0</c> also represent the
+      same <em>number</em>, but they yield different results when converted to
+      string form through <c>float_to_list/1</c>: when given the former it
+      returns a string without a sign, and when given the latter it returns a
+      string with a sign. Therefore, <c>=:=</c> and <c>=/=</c> consider
+      them different.</p>
+
+    <p>The term equivalence operators are useful when reasoning about terms as 
+      opaque values, for example in associative containers or memoized
+      functions where using the equal-to operator (<c>==</c>) can result in
+      subtly incorrect results.</p>
```

Comment:
This is also somewhat confusing. It brings up two new concepts, without explanation or examples (which of course would blow this document out of all proportion). I mean, I know about the subtly incorrect results you mention here, because they did bite me in the past. But I'd say that I wouldn't have been aware of them if I had read this paragraph before :woman_shrugging:

Maybe you should just show briefly how different implementations of the same `$THING` behave differently by using the equality vs term equivalence operators/pattern matching, like `ordsets` (working by equality) and `sets` (working by term equivalence), or the like?
As of late, there is a NOTE box at the top of the [`sets` documentation](https://www.erlang.org/doc/man/sets.html) outlining the differences in behavior between `sets`, `ordsets` and `gb_sets` that you could link to.

Also, I think it is wrong to unconditionally say that using the equality operators can result in incorrect results, it kinda leaves an aftertaste of "equality operators ***bad***!!!". Which they are not: If equality is what you _want_, I'd say it is not wrong.
IMO the real issue is that there is the possibility of a mismatch between how you _expect_ something to behave and how it _really_ behaves if you don't pay attention to equality vs term equivalence, a mismatch that will probably manifest itself every so often, and is hard to find.

Reply:
Reading back on my comment, I realize it sounds a bit harsh :sweat: Sorry for that :pray:

But I realize the wording and maybe-needed verbosity depends to some degree on what assumptions we can make about prior knowledge of a reader.

Reply:
> But I realize the wording and maybe-needed verbosity depends to some degree on what assumptions we can make about prior knowledge of a reader.

If only we could assume that the reader knew everything to begin with... :smile: 

> it kinda leaves an aftertaste of "equality operators bad!!!". Which they are not: If equality is what you want, I'd say it is not wrong.

It's wrong when reasoning about terms as opaque values, which is what that paragraph is about.

Can you try playing around with the wording to see if there's any particular direction you'd prefer (in general, both files if you wish)?

Reply:
> Can you try playing around with the wording to see if there's any particular direction you'd prefer (in general, both files if you wish)?

I guess I had that coming :rofl: Ok, I'll think and play :wink: